get-started/commit-code-and-submit-a-pull-request.md

```diff
@@ -1,11 +1,100 @@
 # Commit code and submit a pull request
 
-including but not limited to:
+The TiDB project uses [Git](https://git-scm.com/) to manage its source code. In order to contribute, you'll need some familiarity with its features so that your changes can be incorporated into the codebase.
 
-* how to organize branches in your local git repository
-* what format needs to be followed for commit message
-* what format needs to be followed for pull request title
-* what content needs to be filled in the pull request description
-* how to request pull request review from reviewers
-* how to get notified when the pull request received review comments
+The goal of this page is to cover some of the more common questions and problems new contributors face. Although some Git basics will be covered here, if you find that this is still a little too fast for you, it might make sense to first read some introductions to Git, such as the Beginner and Getting started sections of [this tutorial from Atlassian](https://www.atlassian.com/git/tutorials/what-is-version-control). GitHub also provides [documentation](https://docs.github.com/en/github/getting-started-with-github/set-up-git) and [guides](https://guides.github.com/introduction/git-handbook/) for beginners, or you can consult the more in depth [book from Git](https://git-scm.com/book/en/v2/).
 
+## Prerequisites
+
+We'll assume that you've installed Git, forked [pingcap/tidb](https://github.com/pingcap/tidb), and cloned the upstream repo to your PC. We'll use the command line interface to interact with Git; there are also a number of GUIs and IDE integrations that can generally do the same things.
+
+If you've cloned the upstream repo, then you will be able to reference it with `origin` in your local repo. Now you should set up a remote for the your fork via
+
+```bash
+git remote add dev https://github.com/username/tidb.git
+```
+
+You can check the remote setting via
+
+```bash
+git remote -v
+# dev    https://github.com/username/tidb.git (fetch)
+# dev    https://github.com/username/tidb.git (push)
+# origin    https://github.com/pingcap/tidb.git (fetch)
+# origin    https://github.com/pingcap/tidb.git (push)
+```
+
+## Standard Process
+
+Below is the normal procedure that you're likely to use for most minor changes and PRs:
+
+1. Ensure that you're making your changes on top of master: `git checkout master`.
+2. Get the latest changes from remote: git pull master.
+3. Make a new branch for your change: `git checkout -b issue-12345-fix`.
+4. Make some changes to the repo and test them.
+5. Stage your changes via `git add .` and then commit them with `git commit`. Of course, making intermediate commits may be a good idea as well. Pay attention on unintentionally commit changes that should not be committed via `git add .`. You can use `git status` to check if the change set is correct.
```

Comment:
```suggestion
5. Stage your changes using `git add .` and then commit the changes using `git commit`. Making intermediate commits might be a good idea too using `git commit -a`, but you need to pay attention to that unintentionally commit changes should not be committed. You can use `git status` to check if the change set is correct.
```

Reply:
`git add .` is different from `git commit -a` on

> new files you have not told Git about are not affected.

I'll make changes based on your suggestion.

Reply:
```suggestion
5. Stage your changes via `git add .` and then commit them with `git commit`. Making intermediate commits might be a good idea as well. Pay attention on unintentionally commit changes that should not be committed via `git add .`. You can use `git status` to check if the change set is correct.
```